### DISCUSSION TITLE
How to set permissions of download directories

### DISCUSSION THREAD

**Discussion by quaintyurt** [slskd/slskd#1014] [Q&A] [answered]
I'm running in a Docker container on unRaid. All of the downloaded files and folders are set with permission `-rw-r-r--`. Is there a setting within the config that changes the default to 777 or to set UMASK?

**Comment by jpdillingham** [comment]
There currently isn't.  I'll convert this to an issue so I can track the request; it'll most likely get rolled up into the "post download automation" milestone with will add support for executing tasks (like shelling out to a script) that will make this easy to implement yourself, but I can probably also build in support directly.

**Comment by bonelifer** [comment]
Having root in Docker instead rootless really makes it hard to post-process files easily. Making it rootless would really make things easier. By being able to set the UID/GID in docker compose:
```
      - PUID=1000
      - PGID=1000
```


There's not a lot of info I could find. Noticed the scripts folder and asked Deepseek about it and it gave me this script. Not sure if that works or the AI is hallucinating.

slskd.yml
```
integration:
  scripts:
    fix_perms:
      on: [DownloadFileComplete]
      run:
        executable: "/bin/bash"
        args: "/app/scripts/fix_perms.sh"
```


```
#!/bin/bash
# File: /app/scripts/fix_perms.sh

# Get the downloaded path from slskd's event data
TARGET=$(echo "$SLSKD_SCRIPT_DATA" | jq -r '.path')

# Verify path exists
if [ ! -e "$TARGET" ]; then
    echo "Error: $TARGET doesn't exist!" >&2
    exit 1
fi

# Change ownership (adjust UID/GID to match your host user)
chown -R 1000:1000 "$TARGET"

# Set safe permissions:
find "$TARGET" -type d -exec chmod 755 {} +  # Folders: rwxr-xr-x
find "$TARGET" -type f -exec chmod 644 {} +  # Files: rw-r--r--

echo "Fixed perms for: $TARGET"
```

Is there some where that documents this integration stuff? Also would you consider supporting rootless docker images?

**Comment by jpdillingham** [comment, marked as answer]
I forgot to follow up on this; #1128 adds a couple of ways to do this.

You can:

* Run the Docker container as a specific user (`--user` or `user:` in compose)
* Set `SLSKD_UMASK` which will set the default umask of the slskd process within the container (in your case, you'd specify 777)
* Set `SLSKD_FILE_PERMISSION_MODE` which will cause slskd to manually set permissions on newly created files

The integration script config generated by AI looks pretty close, I'm surprised the slskd docs have been indexed.  I'm not sure the `jq` path is correct.
